sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -32,9 +43,16 @@
  * HttpClient implementation for OkHttp.
  */
 class OkHttpAsyncHttpClient implements HttpClient {
-    final OkHttpClient httpClient;
-    //
+
     private static final Mono<okio.ByteString> EMPTY_BYTE_STRING_MONO = Mono.just(okio.ByteString.EMPTY);
+    /**
+     * This constant defines a size of Flux based request where buffering in memory becomes less performant
+     * than streaming (which involves thread hops). The value has been established experimentally using
+     * Storage benchmarks.
+     */
+    private static final long BUFFERED_FLUX_REQUEST_THRESHOLD = 100 * 1024L;
```

Comment:
We should check in the benchmark tests as well with this PR. If newer versions of OkHttp require adjusting this number, we just have to run the benchmark tests to update this number.

Reply:
Benchmarks that are helpful are all in the repo in the https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/storage/azure-storage-perf and https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/core/azure-core-perf .
We need real network communication to assess this. Communication over `localhost` isn't good enough when concurrency grows.